### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by ashifkhn**
Auth, API Call, Login and Log Out feature added

### CODE REVIEW

File: src/components/Header.jsx [hunk 1]
```diff
@@ -2,12 +2,29 @@ import React from "react";
...
+  const { login, setLogin } = useAuth();
+
+  const logoutHandler = () => {
+    setLogin(false);
+    dispatch({
+      type: "CLEAR_CART_AND_WISHLIST",
+      payload: [],
+    });
+
+    // navigate("/");
```
Comment (Pratham82) [inline src/components/Header.jsx:25]: Remove logs on line no.6, 25

File: src/pages/ProductsPage/ProductPage.jsx [hunk 2]
```diff
@@ -33,7 +48,7 @@ export const ProductPage = () => {
     return sortedProducts;
   };
 
-  // console.log(sortedProducts);
+  // console.log(datafromApi);
```
Comment (Pratham82) [inline src/pages/ProductsPage/ProductPage.jsx:51]: remove logs

**Review by Pratham82 — APPROVED** [review]
Code stucutre is good, but can be improved mentioning some suggestions:
1. Instead of making the folder name StyleClasses keep it styles
2. In reducer if there is only one reducer, or in future if you are going to create multiple reducers and export from one file, change the name to index.js because it's not a react components and neither we are using any hook here, just the reducer.

File: src/App.js [hunk 3]
```diff
@@ -1,22 +1,42 @@
 import "./App.css";
-import { BrowserRouter, Routes, Route, } from "react-router-dom";
+import { BrowserRouter, Routes, Route, useNavigate, useLocation } from "react-router-dom";
 import Header from "./components/Header";
 import HomePage from "./pages/HomePage/HomePage";
 import { ProductPage } from "./pages/ProductsPage/ProductPage";
 import { CartPage } from "./pages/CartPage/CartPage";
 import { LoginPage } from "./pages/LoginPage/LoginPage";
 import { WishListPAge } from "./pages/WishListPage/WishListPAge";
```
Comment (vikrantkode) [inline src/App.js:8]: import all files from one index.js

File: src/components/Header.jsx [hunk 4]
```diff
@@ -2,12 +2,29 @@ import React from "react";
 import "../StyleClasses/main.css";
 import "./Style/header.css";
 import { Link } from "react-router-dom";
-import { CartState } from "../context/Context";
+import { CartState } from "../context/CartContext";
+// import { useState } from "react";
```
Comment (vikrantkode) [inline src/components/Header.jsx:6]: remove commented code

File: src/components/Sidebar.jsx [hunk 5]
```diff
@@ -8,17 +8,15 @@ export const Sidebar = () => {
...
       <h3 className="p1">Filters</h3>
       <div>
-        <h5>Price</h5>
-        <div>
+        {/* <h5>Price</h5> */}
+        {/* <div>
           <input type="range" className="price-range" min={0} max={20000} />
           <label htmlFor="cowbell">Price</label>
-        </div>
+        </div> */}
```
Comment (vikrantkode) [inline src/components/Sidebar.jsx:19]: remove commented code while pushing

File: src/components/Style/productview.css [hunk 6]
```diff
@@ -1,3 +1,22 @@
...
+right: 10px;
+}
+
+.added{
+  background: rgb(45, 45, 45);
+  padding: 0.3rem;
+  border-radius: 0.5rem;
+  margin: 0.2rem;
+  font-weight: bold;
+  color: white;
```
Comment (vikrantkode) [inline src/components/Style/productview.css:21]: try not using hardcode colors

**Review by vikrantkode — APPROVED** [review]
Code looks good.....some suggestions left..have a look
-Rest good to go

File: src/pages/ProductsPage/ProductPage.jsx (hunk 2 above)
Comment (ashifkhn) [reply src/pages/ProductsPage/ProductPage.jsx:51]: Noted; I'll refactor.

File: src/App.js (hunk 3 above)
Comment (ashifkhn) [reply src/App.js:8]: Noted; I'll refactor.

File: src/components/Header.jsx (hunk 4 above)
Comment (ashifkhn) [reply src/components/Header.jsx:6]: Noted; I'll refactor.

File: src/components/Sidebar.jsx (hunk 5 above)
Comment (ashifkhn) [reply src/components/Sidebar.jsx:19]: Noted; I'll refactor.

File: src/components/Style/productview.css (hunk 6 above)
Comment (ashifkhn) [reply src/components/Style/productview.css:21]: Noted; I'll refactor.

File: src/components/Header.jsx (hunk 1 above)
Comment (ashifkhn) [reply src/components/Header.jsx:25]: Noted; I'll refactor.

**Pull request merged**